src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -56,15 +56,15 @@ public void init() {
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     response.setContentType("application/json");
-    String user = request.getParameter("user");
+    String countryCode = request.getParameter("countryCode");
 
-    if (user == null || user.equals("")) {
```

Comment:
So when messages were posted to the user page, this was checking if the user whose page it was existed, which is (in my head) equivalent to checking if the country whose page it is exists. However, I see your point - I'll add a line to check if the user is logged in, because if they are not then we don't want them posting messages

Reply:
Actually, I just went to add that isUserLoggedIn functionality and it looks like it already exists in messageServlet, lines 80- 83: 
```
if (!userService.isUserLoggedIn()) {
      response.sendRedirect("/index.html");
      return;
    }
```

Does that address your concern?

Reply:
Yeah that looks good!